### PR TITLE
[5.x] Support profile features in CoverageAsPoint

### PR DESCRIPTION
**Pull request by haileyajohnson**
## Description of Changes

Remove time/time unit/time axis as requirement for point feature to address grid as point issues in TDS
Support profile features in CoverageAsPoint
Add tests for CoverageAsPoint

## PR Checklist

- [x] Indicate the version associated with this PR in the Title
       (e.g. "[5.x]: This is my PR title")
- [ ] Link to any issues that the PR addresses
- [ ] Add labels, especially if the PR should be ported to other versions
       (these labels start with "port: ")
- [ ] Open as a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/)
       until ready for review
- [ ] Make sure GitHub tests pass
- [ ] Mark PR as "Ready for Review"

### CODE REVIEW

**Comment by haileyajohnson** [comment]
So this _works_ but it lists 1970-01-01T00:00:00Z as the time unit for any grid as point request on a dataset that doesn't have a time axis. It might be worth adding an `Undefined` type for `CalendarDate` and `CalendarDateUnit`. Open to comments, and we can always change it later.

**Comment by dopplershift** [comment]
Where does that appear as a time unit if you have no time? Is this in e.g. the header for the CSV returns?

**Comment by haileyajohnson** [comment]
@dopplershift the output from what's in this PR, for a grid as point with no time axis, looks like this:

> time,station,latitude[unit="degrees_north"],longitude[unit="degrees_east"],dvs[unit="%"],dvp[unit="%"]
> 1970-01-01T00:00:00Z,GridPointRequestedAt[33.000N_120.000W],33.000,-120.000,0.016,0.06

The `CalendarDate.UNKNOWN` option ends up as 1970-01-01T00:00:00Z instead of just being ignored

**Comment by haileyajohnson** [comment]
I mean the _right way_ I guess would be to leave time out, but that's gonna be a deeper code scour

**Comment by dopplershift** [comment]
Sure. Given that's for the case where the data have no time, this seems fine.

**Review by tdrwenski — APPROVED** [review]

**Comment by haileyajohnson** [comment]
Not ready to merge, realizing this is part of a bigger issue with how we're handling vertical coordinate axes in grid as point...

File: cdm/core/src/main/java/ucar/nc2/ft2/coverage/writer/CoverageAsPoint.java [hunk 1]
```diff
@@ -64,92 +84,333 @@
...
+      if (!wantCovs.isEmpty()) {
+        varGroups.add(new VarGroup(css.getCoordSys().getName(), wantCovs));
       }
     }
   }
 
   public FeatureDatasetPoint asFeatureDatasetPoint() {
-    // for the moment, assume a single station, no vert coord, single lat/lon
-    return new CoverageAsFeatureDatasetPoint(FeatureType.STATION);
+    return varGroups.size() == 0 ? null : new CoverageAsFeatureDatasetPoint();
```
Comment (tdrwenski) [inline cdm/core/src/main/java/ucar/nc2/ft2/coverage/writer/CoverageAsPoint.java]: Maybe a tiny bit nicer:
```suggestion
    return varGroups.isEmpty() ? null : new CoverageAsFeatureDatasetPoint();
```

File: cdm/core/src/main/java/ucar/nc2/ft2/coverage/writer/CoverageAsPoint.java [hunk 2]
```diff
@@ -47,14 +33,48 @@
...
+  private List<VarGroup> varGroups;
   private SubsetParams subset;
   private LatLonPoint latLonPoint;
   private LatLonPoint nearestLatLonPoint;
-  private CalendarDateUnit dateUnit;
+
+  private class VarGroup {
+    private String name;
+    private List<VarData> varData;
+    private FeatureType ftype;
```
Comment (tdrwenski) [inline cdm/core/src/main/java/ucar/nc2/ft2/coverage/writer/CoverageAsPoint.java]: maybe `fType` is nicer?

File: cdm/core/src/test/java/ucar/nc2/ft2/coverage/writer/TestCoverageAsPoint.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    // create coverage as point for all vars in test file
+    gds = CoverageDatasetFactory.open(testFilePath).getCoverageCollections().get(0);
+  }
+
+  @Test
+  public void testVarGroupByDimensions() throws IOException {
+    List<String> varNames = new ArrayList<>();
+    gds.getCoverages().forEach(cov -> {
+      varNames.add(cov.getName());
+    });
```
Comment (tdrwenski) [inline cdm/core/src/test/java/ucar/nc2/ft2/coverage/writer/TestCoverageAsPoint.java]:
```suggestion
    gds.getCoverages().forEach(cov -> varNames.add(cov.getName()));
```

File: cdm/core/src/main/java/ucar/nc2/ft2/coverage/writer/CoverageAsPoint.java [hunk 4]
```diff
@@ -47,14 +33,48 @@
...
+    private CoverageCoordAxis1D zAxis;
+    private String zUnit;
+
+    VarGroup(String name, List<Coverage> wantCovs) throws IOException {
+      this.name = name;
+      this.varData = new ArrayList<>();
+      for (Coverage cov : wantCovs) {
+        this.varData.add(new VarData(cov));
+      }
+      CoverageCoordSys subsetCys = this.varData.get(0).array.getCoordSysForData();
```
Comment (tdrwenski) [inline cdm/core/src/main/java/ucar/nc2/ft2/coverage/writer/CoverageAsPoint.java]: `subsetSys`?

**Review by tdrwenski — APPROVED** [review]

**Pull request merged**